packages/hydrogen/src/cache/create-with-cache.ts

```diff
@@ -6,57 +6,95 @@ import {
   CacheKey,
   runWithCache,
 } from './run-with-cache';
+import {fetchWithServerCache} from './server-fetch';
 import type {WaitUntil} from '../types';
 
 type CreateWithCacheOptions = {
   /** An instance that implements the [Cache API](https://developer.mozilla.org/en-US/docs/Web/API/Cache) */
   cache: Cache;
   /** The `waitUntil` function is used to keep the current request/response lifecycle alive even after a response has been sent. It should be provided by your platform. */
   waitUntil: WaitUntil;
-  /** The `request` object is used to access certain headers for debugging */
-  request?: CrossRuntimeRequest;
+  /** The `request` object is used by the Subrequest profiler, and to access certain headers for debugging */
+  request: CrossRuntimeRequest;
```

Comment:
Since this is required for the Subrequest profiler, and that isn't outlined clearly in the docs, @wizardlyhel and I agreed that this should no longer be optional, and that this PR would be a good place to make that breaking change